checkers: flagName panics

Panic (index-out-of-bounds) hapens on line 36:
```go
c.checkFlagName(call, call.Args[0])
```
It's a mystery how either of these funcs can have 0 args:
```
"flag.Bool", "flag.Duration", "flag.Float64", "flag.String",
		"flag.Int", "flag.Int64", "flag.Uint", "flag.Uint64"
```
Needs investigation (and a fix).

```
flagName: error: runtime error: index out of range
panic: runtime error: index out of range [recovered]
	panic: runtime error: index out of range

goroutine 5110 [running]:
github.com/go-lintpack/lintpack/linter/lintmain/internal/check.(*linter).checkFile.func1.1(0xc00df739e0, 0xc017c3a780)
	$GOPATH/src/github.com/go-lintpack/lintpack/linter/lintmain/internal/check/check.go:136 +0x15b
panic(0x810d20, 0xcbc9e0)
	$GOROOT/src/runtime/panic.go:513 +0x1b7
github.com/go-critic/go-critic/checkers.(*flagNameChecker).VisitExpr(0xc0176c35f0, 0x8cda60, 0xc00a6b55c0)
	$GOPATH/src/github.com/go-critic/go-critic/checkers/flagName_checker.go:36 +0x680
github.com/go-lintpack/lintpack/astwalk.(*exprWalker).WalkFile.func1(0x8c8860, 0xc00a6b55c0, 0xc00657ed01)
	$GOPATH/src/github.com/go-lintpack/lintpack/astwalk/expr_walker.go:23 +0x7f
go/ast.inspector.Visit(0xc00afd8750, 0x8c8860, 0xc00a6b55c0, 0x8c7a20, 0xc00afd8750)
	$GOROOT/src/go/ast/walk.go:373 +0x3a
go/ast.Walk(0x8c7a20, 0xc00afd8750, 0x8c8860, 0xc00a6b55c0)
	$GOROOT/src/go/ast/walk.go:52 +0x66
go/ast.walkExprList(0x8c7a20, 0xc00afd8750, 0xc00657edb0, 0x1, 0x1)
	$GOROOT/src/go/ast/walk.go:26 +0x9e
go/ast.Walk(0x8c7a20, 0xc00afd8750, 0x8c8660, 0xc00a6b5600)
	$GOROOT/src/go/ast/walk.go:207 +0x2103
go/ast.walkStmtList(0x8c7a20, 0xc00afd8750, 0xc00a6b5700, 0x3, 0x4)
	$GOROOT/src/go/ast/walk.go:32 +0x9e
go/ast.Walk(0x8c7a20, 0xc00afd8750, 0x8c87e0, 0xc00a6b0d20)
	$GOROOT/src/go/ast/walk.go:224 +0x1b67
go/ast.Walk(0x8c7a20, 0xc00afd8750, 0x8c9060, 0xc00a68b810)
	$GOROOT/src/go/ast/walk.go:285 +0x2961
go/ast.walkStmtList(0x8c7a20, 0xc00afd8750, 0xc00657eed0, 0x1, 0x1)
	$GOROOT/src/go/ast/walk.go:32 +0x9e
go/ast.Walk(0x8c7a20, 0xc00afd8750, 0x8c87e0, 0xc00a6b0d50)
	$GOROOT/src/go/ast/walk.go:224 +0x1b67
go/ast.Walk(0x8c7a20, 0xc00afd8750, 0x8c8c60, 0xc00a6b0d80)
	$GOROOT/src/go/ast/walk.go:344 +0xd73
go/ast.Inspect(...)
	$GOROOT/src/go/ast/walk.go:385
github.com/go-lintpack/lintpack/astwalk.(*exprWalker).WalkFile(0xc0176c3600, 0xc0022a8680)
	$GOPATH/src/github.com/go-lintpack/lintpack/astwalk/expr_walker.go:21 +0x12c
github.com/go-lintpack/lintpack.(*Checker).Check(...)
	$GOPATH/src/github.com/go-lintpack/lintpack/lintpack.go:126
github.com/go-lintpack/lintpack/linter/lintmain/internal/check.(*linter).checkFile.func1(0xc00df739e0, 0xc0022a8680, 0xc00afef720, 0x1, 0x1, 0x0, 0xc017c3a780)
	$GOPATH/src/github.com/go-lintpack/lintpack/linter/lintmain/internal/check/check.go:144 +0x98
created by github.com/go-lintpack/lintpack/linter/lintmain/internal/check.(*linter).checkFile
	$GOPATH/src/github.com/go-lintpack/lintpack/linter/lintmain/internal/check/check.go:126 +0x120
```

--- a/checkers/flagName_checker.go
+++ b/checkers/flagName_checker.go
@@ -30,12 +30,22 @@ type flagNameChecker struct {
 
 func (c *flagNameChecker) VisitExpr(expr ast.Expr) {
 	call := astcast.ToCallExpr(expr)
-	switch qualifiedName(call.Fun) {
-	case "flag.Bool", "flag.Duration", "flag.Float64", "flag.String",
-		"flag.Int", "flag.Int64", "flag.Uint", "flag.Uint64":
+	sym := astcast.ToIdent(astcast.ToSelectorExpr(call.Fun).Sel)
+	obj := c.ctx.TypesInfo.ObjectOf(sym)
+	if obj == nil {
+		return
+	}
+	pkg := obj.Pkg()
+	if !isStdlibPkg(pkg) || pkg.Name() != "flag" {
+		return
+	}
+
+	switch sym.Name {
+	case "Bool", "Duration", "Float64", "String",
+		"Int", "Int64", "Uint", "Uint64":
 		c.checkFlagName(call, call.Args[0])
-	case "flag.BoolVar", "flag.DurationVar", "flag.Float64Var", "flag.StringVar",
-		"flag.IntVar", "flag.Int64Var", "flag.UintVar", "flag.Uint64Var":
+	case "BoolVar", "DurationVar", "Float64Var", "StringVar",
+		"IntVar", "Int64Var", "UintVar", "Uint64Var":
 		c.checkFlagName(call, call.Args[1])
 	}
 }

--- a/checkers/importShadow_checker.go
+++ b/checkers/importShadow_checker.go
@@ -39,8 +39,7 @@ func (c *importShadowChecker) VisitLocalDef(def astwalk.Name, _ ast.Expr) {
 }
 
 func (c *importShadowChecker) warn(id ast.Node, importedName string, pkg *types.Package) {
-	if pkg.Path() == pkg.Name() {
-		// Сheck for standart library packages.
+	if isStdlibPkg(pkg) {
 		c.ctx.Warn(id, "shadow of imported package '%s'", importedName)
 	} else {
 		c.ctx.Warn(id, "shadow of imported from '%s' package '%s'", pkg.Path(), importedName)

--- a/checkers/testdata/_importable/flag/flag.go
+++ b/checkers/testdata/_importable/flag/flag.go
@@ -1,0 +1,5 @@
+package flag
+
+func Bool() bool {
+	return false
+}

--- a/checkers/testdata/flagName/negative_tests2.go
+++ b/checkers/testdata/flagName/negative_tests2.go
@@ -1,0 +1,7 @@
+package checker_test
+
+import (
+	"github.com/go-critic/go-critic/checkers/testdata/_importable/flag"
+)
+
+var _ = flag.Bool()

--- a/checkers/utils.go
+++ b/checkers/utils.go
@@ -8,6 +8,11 @@ import (
 	"github.com/go-lintpack/lintpack"
 )
 
+// isStdlibPkg reports whether pkg is a package from the Go standard library.
+func isStdlibPkg(pkg *types.Package) bool {
+	return pkg.Path() == pkg.Name()
+}
+
 // isUnitTestFunc reports whether FuncDecl declares testing function.
 func isUnitTestFunc(ctx *lintpack.CheckerContext, fn *ast.FuncDecl) bool {
 	if !strings.HasPrefix(fn.Name.Name, "Test") {

